### ISSUE TITLE
[2.6] Implement generation confidence from token probabilities

### ISSUE THREAD

**Issue by muhit009** [muhit009/VT-Capstone-AI-Trust-Metrics#80]
## User Story
As an Engineering Analyst, I want to know when the AI is uncertain during answer generation, so that I can identify potentially unreliable responses.

## Function
Model-Agnostic Confidence Engine

## Description
Implement generation confidence calculation by extracting and analyzing token-level probabilities (logits) from the LLM during answer generation.

## Methodology
1. During generation, capture token-level logits from the LLM
2. Convert logits to probabilities using softmax
3. Calculate average probability across all generated tokens
4. Higher average probability = model was more certain

## Formula
```
Gen_Confidence = (1/T) × Σ P(token_t | context)
```

Where:
- T = number of tokens in generated answer
- P(token_t | context) = probability of token t given previous context

## Technical Requirements
- Configure LLM (Llama/Mistral) to return logits during generation
- Extract token probabilities from logit outputs
- Calculate average (or geometric mean) across tokens
- Alternative: use minimum token probability as conservative estimate
- Normalize to 0-1 range
- Handle edge cases (very long answers, special tokens)

## Tools/Technologies
- Ollama or vLLM with logits enabled
- Llama-3.1-8B or Mistral-7B
- NumPy for probability calculations
- PyTorch for softmax if needed

## Deliverables
- `generation_confidence.py` - Module implementing generation confidence
- Logit extraction from LLM output
- Probability calculation and aggregation
- Unit tests with sample logits
- Performance analysis (overhead of logit extraction)

## Acceptance Criteria
- [ ] System can extract token probabilities from LLM generation
- [ ] Generation confidence is calculated as average token probability
- [ ] Score interpretation: >0.8 = highly confident, 0.5-0.8 = moderate, <0.5 = uncertain
- [ ] Edge cases handled (empty answers, special tokens)
- [ ] Minimal performance overhead (<10% slowdown)

## Dependencies
- Issue #61 (7.3) - LLM setup with logit access
- Issue #29 (1.3) - fusion rules define how generation confidence is used

## Estimated Time
16 hours

## Priority
High

## Labels
mvp2-engine, high-priority, team-confidence-engine

## Sprint
Sprint 2

**Comment by muhit009** [comment]
Made for local Model
